odoo/addons/base/tests/test_views.py

```diff
@@ -3042,3 +3043,23 @@ def test_views(self):
                 _logger.info('checked %s/%s views', index, len(views))
             with self.subTest(name=view.name):
                 view._check_xml()
+
+@common.tagged('post_install', '-at_install', '-standard', 'render_all_views')
+class TestRenderAllViews(common.TransactionCase):
+
+    @common.users('demo', 'admin')
+    def test_render_all_views(self):
+        env = self.env(context={'lang': 'en_US'})
+        count = 0
+        elapsed = 0
+        for model in env.values():
+            if not model._abstract and model.check_access_rights('read', False):
+                with self.subTest(model=model):
+                    count += 1
+                    elapsed += min(timeit.repeat(
+                        setup='model.invalidate_cache()',  # called each repeat
+                        stmt='model.fields_view_get()',  # measured each number
+                        globals={'model': model},
+                        repeat=5, number=1,  # executed repeat * number
+                    ))
+        _logger.info('Rendered %d views as %s using precisely %ss', count, self.env.user, elapsed)
```

Comment:
That's not a test, why is that a test? Also using timeit just to get aggregate times seems to be rather missing the point.

Reply:
The other solution is to create a sub-command, IMO a test-case `-standard` is fine. Before I added timeit, we had a bare `for _ in range(5)` loop to render each view multiple times and generate better speedscopes. Thanks to timeit, I can also get the precise time spent only in the function and log it. Timeit is nice, both for speedscope (left-heavy) and for the logs.

Reply:
> Thanks to timeit, I can also get the precise time spent only in the function and log it. Timeit is nice, both for speedscope (left-heavy) and for the logs.

You can trivially get the same using precise timers.

The point of timeit is to more easily aggregate calls in order to hide variance and reach a "relevant" total runtime (0.2s using the CLI / `autorange`, pyperf is a bit more complicated but wants a runtime of at least 0.1s) but that's not at all how it's being used here.

Reply:
You're right, let's just use `time.perf_counter()`